DPPP/AddNoiseLBA.cc

```diff
@@ -0,0 +1,166 @@
+#include <string>
+#include <sstream>
+#include <utility>
+#include <vector>
+#include <random>
+
+using namespace casacore;
+
+namespace DP3 {
+  namespace DPPP {
+
+    AddNoiseLBA::AddNoiseLBA (DPInput* input,
+                      const ParameterSet& parset,
+                      const string& prefix)
+    : itsInput(input)
+    {
+      nsteps = parset.getInt (prefix+"nsteps",10);	   
+    }
+
+    AddNoiseLBA::~AddNoiseLBA()
+    {}
+
+    void AddNoiseLBA::updateInfo (const DPInfo& infoIn)
+    {
+      info() = infoIn;
+      info().setNeedVisData();
+      info().setWriteData();
+    }
+
+    void AddNoiseLBA::show (std::ostream& os) const
+    {
+      os << "AddNoiseLBA " << itsName << endl;
+    }
+
+    void AddNoiseLBA::showTimings (std::ostream& os, double duration) const
+    {
+      os << "  ";
+      FlagCounter::showPerc1 (os, itsTimer.getElapsed(), duration);
+      os << " AddNoiseLBA " << itsName << endl;
+    }
+
+    bool AddNoiseLBA::process (const DPBuffer& buf)
+    {
+      double nu;    
+      double stddev;
+      double sefd;
+      itsTimer.start();
+
+      // Name of the column to add the noise (at the moment not used, just a placeholder)
+      string column = "DATA";
+      Array<Complex>::const_contiter indIter = buf.getData().cbegin();
+
+      // Set the exposure
+      double exposure = buf.getExposure();
+
+      // Load the Antenna columns
+      Vector<Int> antenna1 = info().getAnt1();
+      Vector<Int> antenna2 = info().getAnt2();
+      //cout << endl;
+      //cout << "ANTENNA1 " << antenna1.size() << endl;
+      //cout << "ANTENNA2 " << antenna2.size() << endl;
+
+      // Set Number of baselines
+      int n_baselines = antenna1.size();
+      //cout << "Number of baselines = " << n_baselines << endl;
+
+      // Set the number of correlations
+      int n_corr = info().ncorr();
+      //cout << "Number of correlations = " << n_corr << endl;
+
+      // Set the LOFAR_ANTENNA_SET
+      string antennaSet1 = "LBA_OUTER";
+      string antennaSet2 = "LBA_INNER";
+      string antennaSet= getInfo().antennaSet();;
+      
+      // Set the Polynomial coefficients
+      double * coeff;
+      if (antennaSet.compare(antennaSet1)) coeff = coeffs_outer;
+      if (antennaSet.compare(antennaSet2)) coeff = coeffs_inner;
+
+      //int nant = getInfo().antennaNames().size();
+      //cout << "N antennas " << nant << endl;
+
+      // Set the number of frequency channels
+      int n_freq = getInfo().chanFreqs().size();
+      //cout << "Number of frequencies = " << n_freq << endl;
+      Vector<Double> chan_freq = getInfo().chanFreqs();
+      Vector<Double> chan_width = getInfo().chanWidths();
+
+      unsigned seed = std::chrono::system_clock::now().time_since_epoch().count();
+      std::default_random_engine generator (seed);
+
+      for (int icorr=1; icorr<n_corr; icorr++)
+      {
+        for (int ifreq=0; ifreq<n_freq; ifreq++) 
+        {
+          nu = chan_freq(ifreq);
+	  sefd = coeff[0]+
```

Comment:
At some lines there's inconsistency in indentation. Once #276 this can automatically be fixed with a script, so no need to really worry about it.